### PR TITLE
Update dependency target and type mapping

### PR DESCRIPTION
**Pull request by vishweshbankwar**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs [hunk 1]
```diff
@@ -175,7 +175,7 @@ internal static string GetDependencyUrl(this AzMonList tagObjects)
             var httpScheme = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpScheme)?.ToString();
             if (!string.IsNullOrWhiteSpace(httpScheme))
             {
-                string defaultPort = GetDefaultPort(httpScheme);
+                string defaultPort = GetDefaultHttpPort(httpScheme);
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:178]: nit: you may move this line to after the httpTarget null check :)

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs [hunk 2]
```diff
@@ -275,5 +287,95 @@ internal static string GetUrlUsingHostAndPort(this AzMonList tagObjects, string
...
+                    else
+                    {
+                        target = httpHost;
+                    }
+                    return target;
+                }
+                var httpUrl = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpUrl)?.ToString();
+                if (!string.IsNullOrEmpty(httpUrl) && Uri.TryCreate(httpUrl.ToString(), UriKind.RelativeOrAbsolute, out var uri) && uri.IsAbsoluteUri)
+                {
+                    target = uri.Host;
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:354]: We could return `Uri.Authority` instead of writing below logic.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs [hunk 3]
```diff
@@ -275,5 +287,95 @@ internal static string GetUrlUsingHostAndPort(this AzMonList tagObjects, string
...
+            switch (type)
+            {
+                case PartBType.Http:
+                    defaultPort = GetDefaultHttpPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpScheme)?.ToString());
+                    break;
+                case PartBType.Db:
+                    defaultPort = GetDefaultDbPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeDbSystem)?.ToString());
+                    break;
+                default:
+                    defaultPort = "0";
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:323]: DefaultPort will not be used in setting telemetry properties. it is only used for comparison. We will always use the port from activity tags.
So "0" will never be exported.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs [hunk 4]
```diff
@@ -275,5 +287,95 @@ internal static string GetUrlUsingHostAndPort(this AzMonList tagObjects, string
...
+            var peerService = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributePeerService)?.ToString();
+            if (!string.IsNullOrEmpty(peerService))
+            {
+                target = peerService;
+                return target;
+            }
+
+            if (type == PartBType.Http)
+            {
+                var httpHost = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpHost)?.ToString();
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:336]: Reviewed the source for complete file. Looks like we are reading `SemanticConventions.AttributeHttpHost` from the list at 2 different places for a single telemetry(`GetUrlUsingHttpHost` and `GetDependencyTarget`). Same for few other attributes too. Is it possible to refactor to read from the list once?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs:336]: That is a good point - will do some refactoring in separate PR.

**Pull request merged**